scipy/io/tests/test_mmio.py

```diff
@@ -387,6 +387,20 @@ def test_sparse_formats(self):
                 result = mmread(fn).todense()
                 assert_array_almost_equal(result, expected)
 
+    def test_precision(self):
+        for precision in range(1, 8):
+            # construct sparse matrix with pi as last main diagonal entry
+            n = 10**precision + 1
+            A = scipy.sparse.lil_matrix((n, n))
+            A[n-1, n-1] = pi
+            # write matrix with low precision and read again
+            mmwrite(self.fn, A, precision=precision)
+            A = scipy.io.mmread(self.fn)
+            # check for right entries in matrix
+            assert_array_equal(A.row, [n-1])
+            assert_array_equal(A.col, [n-1])
+            assert_array_almost_equal(A.data, [pi], decimal=precision-1)
```

Comment:
I see what you mean, but I see that the Fortran code does something similar, using Fortran's automatic formatting for float / complex, e.g.:

```
write(unit=ounit,fmt=*)indx(i),jndx(i),rval(i)
```

in http://math.nist.gov/MatrixMarket/mmio/f/mmio.f. This seems to invoke "list-directed formatting" - which I think gives exponential format too - see : http://docs.oracle.com/cd/E19957-01/805-4939/6j4m0vnc5/index.html

For example, `real * 8` (float64) gets format `1PE22.13.E2`.  I can't immediately see what that means - but it looks like it's an exponential format at least.